src/PowerShellEditorServices/Session/Host/TerminalPSHostRawUserInterface.cs

```diff
@@ -184,11 +145,87 @@ public override Size MaxWindowSize
         /// <returns>A KeyInfo struct with details about the current keypress.</returns>
         public override KeyInfo ReadKey(ReadKeyOptions options)
         {
-            Logger.Write(
-                LogLevel.Warning,
-                "PSHostRawUserInterface.ReadKey was called");
+            // Converts ConsoleKeyInfo objects to KeyInfo objects and caches key down events
+            // for the next key up request.
+            KeyInfo ProcessKey(ConsoleKeyInfo key, bool isDown)
```

Comment:
I don't think we've been using nested functions much... and I don't think it needs to be nested does it? Maybe consider pulling it out into a private function.

Reply:
Nah doesn't need to be.  Just another way to limit scope and noise, I can take it out though.